lib/parse.js

```diff
@@ -177,12 +177,40 @@ module.exports = {
       cache: [],
       externals: [],
       externalRefs: {},
+      ignoreIOErrors: true,
       rewriteRefs: true,
       openapi: openapi,
       files: files
     });
   },
 
+  /**
+   * Resolves remote and URL $ref from OpenAPI definition based on given options
+   *
+   * @param {*} openapi - OpenAPI definition
+   * @param {*} options - options
+   * @param {*} cb - callback function
+   * @return {*} - err if present
+   */
+  resolveRemoteRefs: function (openapi, options, cb) {
+    if (options.resolveRemoteRefs) {
+      return resolver.resolve(openapi, options.sourceUrl, {
```

Comment:
@VShingala Is there place in schema itself where the user can specify this sourceUrl? If yes then we should also check if that is present.